### ISSUE TITLE
hexpm returns responds successfully when no docs storage/server is configured

### ISSUE THREAD

**Issue by starbelly** [hexpm/hexpm#923]
Currently hexpm will respond with a 201 or 204 (depending on context) when no docs storage/server is configured. The server will always respond with 201 on publish and respond with 204 on revert. 

 The 204 case is not wrong, but I'm not sure this is a good signal when the issue is more incessant. It's conflicting because this may be the best status to return for hex.pm but perhaps not for private hexpm instances. 

The 201 case however seems more dubious in that, nothing was actually modified, not really. The audit_log is updated and packages.docs_updated_at  is updated, but I'm not sure that in and of itself is deserving of a 201 in and of itself. Once again, conflicting as this may be the best option for hex.pm, but maybe not the best for private hexpm instances.

These issues came to light from doing local testing with a local hexpm instance.

**Comment by wojtekmach** [comment]
> when no docs storage/server is configured

how did you determine no storage is configured? Even in dev, we use the Hexpm.Store.Local storage adapter, if we don't (or if we have empty adapter somehow) that's definitely a bug and we should crash.

**Comment by starbelly** [comment]
@wojtekmach Let me look at this further tomorrow, it may be a misunderstanding on my end.

**Comment by ericmj** [comment]
What endpoints or URLs are returning these status codes? hexpm is not really supposed to work if no storage is configured and we do not support private repos in this project so I am not sure this is something we should fix. I would expect it to crash and return 500 if storage is misconfigured.

**Comment by starbelly** [comment]
@ericmj @wojtekmach Totally fair. I just wasn't sure. 

The end point that returns `nil` is `/packages/:name/releases/:version/docs` and only on delete. It's set to return an empty string, so that makes sense after digging more. 
  
 I didn't realize at the time of writing this that there is a local adapter, I made an assumption because if I'm doing local testing and publish docs and I go to my local instance, then it appears as if nothing was ever uploaded, this is why the 201 response confused me. All this Makes sense and I think I just had a good bit of confusion on my end about the flow in local dev mode.  In retrospect I think I got so deep in the weeds doing QA that I couldn't tell the forest from the trees for a minute and thus this is starting to feel like a non-issue :)   

 I think to do better QA I'd need to set up something close to a production instance or send some PRs up to hexpm to closer mimic what happens in prod with docs ( then again there still may be more that I'm missing from my local setup for that). Regardless, as stated this feels like a non-issue now.
